### PR TITLE
Testdrive: Use the configured kafka address

### PR DESCRIPTION
**Pull request by cirego**
When asserting kafka broker RTT stats, use the configured
broker address, not a hardcode hostname / port combination.

Fxies #6459

### CODE REVIEW

**Comment by philip-stoev** [comment]
Unfortunately in testdrive the kafka address is localhost, but the hostname is something else. This patch seemed to do the right thing for me:

```
diff --git a/src/testdrive/Cargo.toml b/src/testdrive/Cargo.toml
index 54a743b4a..3f098ef5c 100644
--- a/src/testdrive/Cargo.toml
+++ b/src/testdrive/Cargo.toml
@@ -19,6 +19,7 @@ flate2 = "1.0.20"
 futures = "0.3.13"
 interchange = { path = "../interchange" }
 itertools = "0.9.0"
+hostname = "^0.3"
 kafka-util = { path = "../kafka-util" }
 krb5-src = { version = "0.2.3", features = ["binaries"] }
 lazy_static = "1.4.0"
diff --git a/src/testdrive/src/action.rs b/src/testdrive/src/action.rs
index 10d7caf4b..674f613ea 100644
--- a/src/testdrive/src/action.rs
+++ b/src/testdrive/src/action.rs
@@ -383,6 +383,11 @@ pub fn build(cmds: Vec<PosCommand>, state: &State) -> Result<Vec<PosAction>, Err
         state.materialized_user.clone(),
     );
 
+    vars.insert(
+        "os.hostname".into(),
+        hostname::get().unwrap().to_string_lossy().to_string()
+    );
+
     for cmd in cmds {
         let pos = cmd.pos;
         let wrap_err = |e| InputError { msg: e, pos };
diff --git a/test/testdrive/kafka-stats.td b/test/testdrive/kafka-stats.td
index 785beb43d..8d19ea9a2 100644
--- a/test/testdrive/kafka-stats.td
+++ b/test/testdrive/kafka-stats.td
@@ -97,10 +97,8 @@ rx_msgs
 # RTT metrics are all timing dependent and there's no method to assert on the values.
 # Just assert that our source created a metrics object for both the broker and group coordinator
 # Ignore the text after the '/' because that is autogenerated and not deterministic
-> SELECT split_part(broker_name, '/', 1) FROM mz_kafka_broker_rtt ORDER BY broker_name;
-split_part
------------
-"${testdrive.kafka-addr}"
+> SELECT split_part(broker_name, '/', 1) = '${testdrive.kafka-addr}' OR position(lower('${os.hostname}') IN lower(split_part(broker_name, '/', 1))) != 0 FROM mz_kafka_broker_rtt ORDER BY broker_name;
+true
 
 # Drop the sources and verify that metrics have been removed
 > DROP VIEW test1
```

**Comment by cirego** [comment]
> Unfortunately in testdrive the kafka address is localhost, but the hostname is something else. This patch seemed to do the right thing for me:
> 
> ```
> diff --git a/src/testdrive/Cargo.toml b/src/testdrive/Cargo.toml
> index 54a743b4a..3f098ef5c 100644
> --- a/src/testdrive/Cargo.toml
> +++ b/src/testdrive/Cargo.toml
> @@ -19,6 +19,7 @@ flate2 = "1.0.20"
>  futures = "0.3.13"
>  interchange = { path = "../interchange" }
>  itertools = "0.9.0"
> +hostname = "^0.3"
>  kafka-util = { path = "../kafka-util" }
>  krb5-src = { version = "0.2.3", features = ["binaries"] }
>  lazy_static = "1.4.0"
> diff --git a/src/testdrive/src/action.rs b/src/testdrive/src/action.rs
> index 10d7caf4b..674f613ea 100644
> --- a/src/testdrive/src/action.rs
> +++ b/src/testdrive/src/action.rs
> @@ -383,6 +383,11 @@ pub fn build(cmds: Vec<PosCommand>, state: &State) -> Result<Vec<PosAction>, Err
>          state.materialized_user.clone(),
>      );
>  
> +    vars.insert(
> +        "os.hostname".into(),
> +        hostname::get().unwrap().to_string_lossy().to_string()
> +    );
> +
>      for cmd in cmds {
>          let pos = cmd.pos;
>          let wrap_err = |e| InputError { msg: e, pos };
> diff --git a/test/testdrive/kafka-stats.td b/test/testdrive/kafka-stats.td
> index 785beb43d..8d19ea9a2 100644
> --- a/test/testdrive/kafka-stats.td
> +++ b/test/testdrive/kafka-stats.td
> @@ -97,10 +97,8 @@ rx_msgs
>  # RTT metrics are all timing dependent and there's no method to assert on the values.
>  # Just assert that our source created a metrics object for both the broker and group coordinator
>  # Ignore the text after the '/' because that is autogenerated and not deterministic
> -> SELECT split_part(broker_name, '/', 1) FROM mz_kafka_broker_rtt ORDER BY broker_name;
> -split_part
> ------------
> -"${testdrive.kafka-addr}"
> +> SELECT split_part(broker_name, '/', 1) = '${testdrive.kafka-addr}' OR position(lower('${os.hostname}') IN lower(split_part(broker_name, '/', 1))) != 0 FROM mz_kafka_broker_rtt ORDER BY broker_name;
> +true
>  
>  # Drop the sources and verify that metrics have been removed
>  > DROP VIEW test1
> ```

Hi @philip-stoev, thanks for the detail! It's a bummer that the `broker_name` in the broker stats structure doesn't match the broker name specified in the `KAFKA` create statement. Can you share the Kafka broker configuration that you're running with? I'm wondering if our consumer is dynamically discovering brokers from your configuration.

Should we consider removing this testcase instead?

**Comment by philip-stoev** [comment]
I am using the confluent dev environment, the one you get when you run ```confluent local services start```.

Removing the test case is totally fine with me , it is too much trouble to get right. You may wish to simply check that the table column in question ```IS NOT NULL``` and is not an empty string and that would be enough.

**Comment by cirego** [comment]
> I am using the confluent dev environment, the one you get when you run `confluent local services start`.
> 
> Removing the test case is totally fine with me , it is too much trouble to get right. You may wish to simply check that the table column in question `IS NOT NULL` and is not an empty string and that would be enough.

Hi @philip-stoev, I went ahead and implented the "not NULL and not empty string" check. Please let me know what you think and thanks for helping with this!

**Review by philip-stoev — APPROVED** [review]
Perfect, thanks!

File: test/testdrive/kafka-stats.td [hunk 1]
```diff
@@ -97,10 +97,10 @@ rx_msgs
 # RTT metrics are all timing dependent and there's no method to assert on the values.
 # Just assert that our source created a metrics object for both the broker and group coordinator
 # Ignore the text after the '/' because that is autogenerated and not deterministic
```
Comment (benesch) [inline test/testdrive/kafka-stats.td:99]: This last line of the comment looks stale now!

**Pull request merged**